src/engine/console.cpp

```diff
@@ -521,62 +594,97 @@ bool consolekey(int code, bool isdown)
                 break;
 
             case SDLK_RIGHT:
-                if(commandpos>=0 && ++commandpos>=(int)strlen(commandbuf)) commandpos = -1;
+                if (commandpos >= 0 && ++commandpos >= (int)strlen(commandbuf)) 
+                {
+                    commandpos = -1;
+                }
                 break;
 
             case SDLK_UP:
-                if(histpos > history.length()) histpos = history.length();
-                if(histpos > 0) history[--histpos]->restore();
+                if (histpos > history.length())
+                {
+                    histpos = history.length();
+                }
```

Comment:
Use std::min or similiar.

Reply:
I don't touch that code, it is unrelated to this PR I just made it follow our code guidelines, a little confusing Ik, that's why I stopped doing it in my other PRs